### DISCUSSION TITLE
Manage Vercel's secrets from the code?

### DISCUSSION THREAD

**Discussion by jibla** [vercel/community#3206] [Help] [answered]
### Problem Description

Hello,

I have a straightforward query. Is it a good practice or an antipattern to manage secrets, such as environment variables, directly from the code? By this, I mean the capability to dynamically add or update an environment variable from the code that benefits from these secrets.

Moreover, is this even permissible in Vercel?

**Comment by amyegan** [comment, marked as answer]
Hi @jibla! It sounds like [Vercel Edge Config](https://vercel.com/docs/storage/edge-config#) is perfect for your situation.

It is also possible to [update an environment variable from code via the API](https://vercel.com/docs/rest-api/endpoints#edit-an-environment-variable), but the project would need to be redeployed before the new values would take effect.

**Reply by jibla** [reply]
Thank you @amyegan 

I have been contemplating the use of Edge Config. However, I am working with environmental data (tokens) which necessitates constant access to the most up-to-date information, as well as the assurance that the data will not be lost from the storage system. It's also critical for my use-case that I am able to manage these tokens, adding or removing them on the fly from the application that utilizes them. 

From my current understanding, this kind of storage seems to be geared more towards temporary or non-essential data, functioning in a capacity similar to that of a cache. 

Right now, I use KV storage, which I understand is not effective in my case and therefore will be much more expensive.

I missing anything or misunderstanding the use-case for this Edge Config?
